runtime/src/nonce_keyed_account.rs

```diff
@@ -102,6 +113,16 @@ impl<'a> NonceKeyedAccount for KeyedAccount<'a> {
     ) -> Result<(), InstructionError> {
         let merge_nonce_error_into_system_error = invoke_context
             .is_feature_active(&feature_set::merge_nonce_error_into_system_error::id());
+
+        if invoke_context.is_feature_active(&nonce_must_be_writable::id()) && !self.is_writable() {
```

Comment:
This check seems fine to add but in most cases it the instruction would fail due to the runtime checks anyways. Technically it was possible to use a read-only nonce account when `lamports == 0`, though.

Reply:
Yeah, same comment as above